src/OrchardCore.Modules/OrchardCore.Facebook/Endpoints/GetSdkEndpoints.cs

```diff
@@ -0,0 +1,137 @@
+using System.IO.Hashing;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.Caching.Memory;
+using OrchardCore.Facebook.Settings;
+using OrchardCore.Settings;
+
+#nullable enable
+
+namespace OrchardCore.Facebook.Endpoints;
+
+public static class GetSdkEndpoints
+{
+    public static IEndpointRouteBuilder AddSdkEndpoints(this IEndpointRouteBuilder builder)
+    {
+        builder.MapGet("/OrchardCore.Facebook/sdk/{hash}/init.js", GetInitScriptEndpoint.HandleRequestAsync)
+            .AllowAnonymous()
+            .DisableAntiforgery();
+
+        builder.MapGet("/OrchardCore.Facebook/sdk/{hash}/sdk.{culture:length(2,6)}.js", GetFetchScriptEndpoint.HandleRequestAsync)
+            .AllowAnonymous()
+            .DisableAntiforgery();
+
+        return builder;
+    }
+
+    public static class GetInitScriptEndpoint
+    {
+        // Update this version when the script changes to invalidate client caches
+        private static readonly int ScriptVersion = 1;
+
+        public static ulong HashCacheBustingValues(FacebookSettings settings)
+        {
+            return XxHash3.HashToUInt64(Encoding.UTF8.GetBytes(
+                        String.Concat(
+                            ScriptVersion.ToString(),
```

Comment:
`ToString` on a number needs to be culture invariant when used for technical purposes, or you would get different string for some cultures.

```suggestion
                            ScriptVersion.ToString(CultureInfo.InvariantCulture),
```